cluster/src/dunit/scala/org/apache/spark/sql/TPCHDUnitTest.scala

```diff
@@ -77,6 +78,259 @@ class TPCHDUnitTest(s: String) extends ClusterManagerTestBase(s)
+          prepStatement.setString(6, parameters(5))
+          prepStatement.setString(7, "Brand#" + parameters(6))
+          prepStatement.setString(8, parameters(7))
+          prepStatement.setString(9, parameters(8))
+        }
+        case 20 => {
+          println("Executing query#20")
+          prepStatement = conn.prepareStatement(TPCH_Queries.getQuery20ForPrepStatement)
+          val parameters = TPCH_Queries.getQ20Parameter(isDynamic)
+          prepStatement.setString(1, parameters(0) + "%")
+          prepStatement.setString(2, parameters(1))
+          prepStatement.setString(3, parameters(2))
+          prepStatement.setString(4, parameters(3))
+        }
+        case 21 => {
+          println("Executing query#21")
+          prepStatement = conn.prepareStatement(TPCH_Queries.getQuery21ForPrepStatement)
+          var parameters = TPCH_Queries.getQ21Parameter(isDynamic)
+          prepStatement.setString(1, parameters(0))
+        }
+        case 22 => {
+          println("Executing query#22")
+          prepStatement = conn.prepareStatement(TPCH_Queries.getQuery22ForPrepStatement)
+          var parameters = TPCH_Queries.getQ22Parameter(isDynamic)
+          prepStatement.setString(1, parameters(0))
+          prepStatement.setString(2, parameters(1))
+          prepStatement.setString(3, parameters(2))
+          prepStatement.setString(4, parameters(3))
+          prepStatement.setString(5, parameters(4))
+          prepStatement.setString(6, parameters(5))
+          prepStatement.setString(7, parameters(6))
+          prepStatement.setString(8, parameters(7))
+          prepStatement.setString(9, parameters(8))
+          prepStatement.setString(10, parameters(9))
+          prepStatement.setString(11, parameters(10))
+          prepStatement.setString(12, parameters(11))
+          prepStatement.setString(13, parameters(12))
+          prepStatement.setString(14, parameters(13))
+        }
+      }
+      if (prepStatement != null) {
+        val rs = prepStatement.executeQuery()
+        verifyTPCHQueryResult(rs, s"Snappy_$query.out")
+        rs.close()
+        prepStatement.close()
+      }
+
+      }
+    // scalastyle:on println
+    }
+
+  private def verifyTPCHQueryResult(rs: ResultSet, expectedResultFile: String): Unit = {
+    val rsmd = rs.getMetaData
+    val columnsNumber = rsmd.getColumnCount
+    var count = 0
+    val result = scala.collection.mutable.ArrayBuffer.empty[String]
+    while (rs.next()) {
+      count += 1
+      var row: String = ""
+      for (i <- 1 to columnsNumber) {
+        if (i > 1) row += ","
+        row = row + rs.getString(i)
+      }
+      result += row
+    }
+    // scalastyle:off println
+    println(s"Number of rows : $count")
+    // scalastyle:on println
+
+    val expectedFile = sc.textFile(getClass.getResource(
+      s"/TPCH/RESULT/$expectedResultFile").getPath)
+    val expectedNoOfLines = expectedFile.collect().size
+    assert(count == expectedNoOfLines)
```

Comment:
Just dump the actual results in a csv and may be you can compare the actual results as well.

Reply:
Done